packages/nimble-components/src/rich-text/editor/tests/rich-text-editor-mention.spec.ts

```diff
@@ -0,0 +1,877 @@
+            testMention.appendChild(mappingUser);
+        });
+    }
+    element.appendChild(testMention);
+    await waitForUpdatesAsync();
+}
+
+describe('RichTextEditorMention', () => {
+    let element: RichTextEditor;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let pageObject: RichTextEditorPageObject;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+        await connect();
+        pageObject = new RichTextEditorPageObject(element);
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('should update "empty" when the mention is loaded with "setMarkdown"', async () => {
+        expect(element.empty).toBeTrue();
+
+        element.setMarkdown('<user:1>');
+
+        await appendUserMentionConfiguration(element);
+        expect(element.empty).toBeFalse();
+
+        element.setMarkdown('');
+        expect(element.empty).toBeTrue();
+    });
+
+    it('Should return same markdown when mention markdown string is assigned with valid configuration element', async () => {
+        element.setMarkdown('<user:1>');
+        await appendUserMentionConfiguration(element);
+        expect(element.getMarkdown()).toBe('<user:1>');
+    });
+
+    it('Should return same markdown when mention markdown string is assigned without configuration element', () => {
+        element.setMarkdown('<user:1>');
+        expect(element.getMarkdown()).toBe('<user:1>');
+    });
+
+    it('Should return same markdown for assigned mention markdown when removing configuration element in the same editor', async () => {
+        element.setMarkdown('<user:1>');
+        await appendUserMentionConfiguration(element);
+        expect(element.getMarkdown()).toBe('<user:1>');
+
+        const renderedUserMention = element.lastElementChild as RichTextMentionUsers;
+        element.removeChild(renderedUserMention);
+        await waitForUpdatesAsync();
+        expect(element.getMarkdown()).toBe('<user:1>');
+    });
+
+    describe('user mention dynamic loading', () => {
+        it('adding mention configuration converts the absolute link matching the pattern to mention node', async () => {
+            element.setMarkdown('<user:1>');
+
+            expect(pageObject.getEditorTagNames()).toEqual(['P', 'A']);
+            expect(pageObject.getEditorLeafContents()).toEqual(['user:1']);
+            await appendUserMentionConfiguration(element);
+
+            expect(pageObject.getMarkdownRenderedTagNames()).toEqual([
+                'P',
+                `${richTextMentionUsersViewTag}`.toUpperCase(),
+                'BR'
+            ]);
+            expect(
+                pageObject.getEditorMentionViewAttributeValues('mention-label')
+            ).toEqual(['1']);
+        });
+
+        // TODO: Once the rich text validator added for duplicate configuration elements, below test case should be updated
```

Comment:
Can you link to a PR or issue associated with this TODO?

Reply:
Added the PR link